04-onion-routing.md

```diff
@@ -164,91 +161,29 @@ It is 1300 bytes long and has the following structure:
+### `payload` format
 
-## Legacy `hop_data` payload format
+This is formatted according to the Type-Length-Value format defined in [BOLT #1](01-messaging.md#type-length-value-format).
 
-The `hop_data` format is identified by a single `0x00`-byte length, for backward compatibility.
-Its payload is defined as:
-
-1. type: `hop_data` (for `realm` 0)
-2. data:
-   * [`short_channel_id`:`short_channel_id`]
-   * [`u64`:`amt_to_forward`]
-   * [`u32`:`outgoing_cltv_value`]
-   * [`12*byte`:`padding`]
-
-Field descriptions:
-
-   * `short_channel_id`: The ID of the outgoing channel used to route the 
-      message; the receiving peer should operate the other end of this channel.
-
-   * `amt_to_forward`: The amount, in millisatoshis, to forward to the next
-     receiving peer specified within the routing information.
-
-     For non-final nodes, this value amount MUST include the origin node's computed _fee_ for the
-     receiving peer. When processing an incoming Sphinx packet and the HTLC
-     message that it is encapsulated within, if the following inequality doesn't hold,
-     then the HTLC should be rejected as it would indicate that a prior hop has
-     deviated from the specified parameters:
-
-          incoming_htlc_amt - fee >= amt_to_forward
-
-     Where `fee` is calculated according to the receiving peer's advertised fee
-     schema (as described in [BOLT #7](07-routing-gossip.md#htlc-fees)).
-
-     For the final node, this value MUST be exactly equal to the incoming htlc
-     amount, otherwise the HTLC should be rejected.
-
-   * `outgoing_cltv_value`: The CLTV value that the _outgoing_ HTLC carrying
-     the packet should have.
-
-          cltv_expiry - cltv_expiry_delta >= outgoing_cltv_value
-
-     Inclusion of this field allows a hop to both authenticate the information
-     specified by the origin node, and the parameters of the HTLC forwarded,
-     and ensure the origin node is using the current `cltv_expiry_delta` value.
-     If there is no next hop, `cltv_expiry_delta` is 0.
-     If the values don't correspond, then the HTLC should be failed and rejected, as
-     this indicates that either a forwarding node has tampered with the intended HTLC
-     values or that the origin node has an obsolete `cltv_expiry_delta` value.
-     The hop MUST be consistent in responding to an unexpected
-     `outgoing_cltv_value`, whether it is the final node or not, to avoid
-     leaking its position in the route.
-
-   * `padding`: This field is for future use and also for ensuring that future non-0-`realm`
-     `hop_data`s won't change the overall `hop_payloads` size.
-
-When forwarding HTLCs, nodes MUST construct the outgoing HTLC as specified
-within `hop_data` above; otherwise, deviation from the specified HTLC
```

Comment:
Wouldn't it be useful to keep this information?

Reply:
I can rework it. The Old format mixed requirements with discussion: I moved the requirements, but should probably put the discussion somewhere...